### DISCUSSION TITLE
Questions regarding Aqara Radiator Thermostat E1 (SRTS-A01)

### DISCUSSION THREAD

**Discussion by adynis** [Koenkk/zigbee2mqtt#15557] [Devices]
Hi,
I'm testing Aqara TRV (SRTS-A01) in Home Assistant using Zigbe2mqtt. I just setup an external sensor and it seems to read correctly the external temp (I find it correctly updated as "local_temperature" in the "State" page)
But I have some questions and I don't know where to ask / find answer:

1. What's the difference between preset==manual and preset==auto ? (what should I use?)
2. Can the TRV have intermediate "values" of the valve (like ... 75% opened) , or it can be only fully closed or fully opened ? (some times I heard the motor working just a little bit, like it closed it or opened it just a little bit)
3. Do I need to set an automation to open the Valve / close the valve depending on temperature ? (it seems to make it by itself automatically in both preset=manual or preset=auto ... and this makes me confused ... )
4. How can I read the current value (Valve opened, Valve closed ) ? (I tested by warming the external sensor, I heard the valve motor, the radiator got cold (so it worked) but in the State page I still see: "state": "ON", "system_mode": "heat" ...)
5. I've updated the firmware version from aqara app to ...0925. But in zigbe2mqtt I still see "0.0.0_0025", is this ok ?
6. I had today "voltage=3100" with "battery=100" and later in the same day: "voltage=3100" but with "battery=82" ... How is this possible ?

Thanks,

**Comment by metawops** [comment]
How did you manage to set which external temperature sensor to link to the E1 Thermostat? I have several of these E1 devices here and already saw via HA's history that the actual room temperature (measured with an Aqara Temp sensor) is lower than the temperature measured by the E1's built-in sensor. So I'd like to link the external temp sensor with the E1 and not use the built-in sensor anymore. But how is this done? I can only find a setting that switches between "external" or "internal" but no setting where I could select my external temp sensor to use ... 😳

**Reply by protyposis** [reply]
https://github.com/Koenkk/zigbee2mqtt/discussions/14291

**Comment by protyposis** [comment]
I am also wondering what the difference of the presents is supposed to be. It seems that it's not related to whether a programmed schedule is active (see https://github.com/Koenkk/zigbee-herdsman-converters/pull/5058#issuecomment-1334236460).

 2. Yes, it's not just a binary open/closed switch but can have intermediate levels, that is why you hear the short motor noises. The valve level/percentage is not (yet?) exposed.
 3. Not necessary, as this is the core feature of every programmable/smart thermostat. You set the target temperature and it handles the rest. You can however still write automations for more advanced use cases but this is kind of hacky as you cannot control the valve directly - only indirectly through the target temperature. See https://github.com/KartoffelToby/better_thermostat/ for an advanced automation.
 4. The system mode "heat" can also be read as "heat when necessary" or "retain target temperature". It is the operating mode the thermostat is in, and is not meant to tell you when it's actually heating. As written in point 2, the valve can have many different opening levels, and which one do you consider "heating"? For example, the valve can be opened a bit, just not enough to keep the room at the target temperature, but the radiator will still be warm. Is this heating? So this isn't super trivial to answer, but a practical heuristic can be calculated from the room temperature and thermostat setpoint: it is heating when the room temperature is below the setpoint.
 5. Yes this is "ok", what you see is basically a placeholder firmware version. The actual version is reported via a proprietary message which hasn't been implemented in Z2M yet. See https://github.com/Koenkk/zigbee2mqtt/issues/13993#issuecomment-1348618655.
 6. If this really happened, it must be a bug. The thermostat does not report battery percentage, but the percentage is directly calculated from the voltage, and always considered 100% above 3V. See https://github.com/Koenkk/zigbee-herdsman-converters/blob/116ebb05fe782b980ab29ac0b867a05bb8447b18/devices/xiaomi.js#L2695.

**Reply by adynis** [reply]
Thanks for the detailed answer ! 👍

**Comment by groenmarsmannetje** [comment]
There are more people who are confused by the manual/auto presets. This is very confusing and should be solved in my opinion. 

#19070

**Comment by brunialti** [comment]
One more point
I’m using valves by Aqara. While configuring the internal scheduler, I noticed that it operates with a 1-hour difference compared to the hosting hub. For example, if I schedule the temperature to change at 14:00, the scheduler starts the action at 15:00. The Z2m Is in a container. I've checked both container's and host' datetime and found them correct. Any suugestion ?
